sotodlib/mapmaking/demod_mapmaker.py

```diff
@@ -0,0 +1,254 @@
+import numpy as np
+from pixell import enmap, utils, tilemap, bunch
+import so3g.proj
+
+from .. import coords
+from .utilities import *
+from .pointing_matrix import *
+
+class DemodMapmaker:
+    def __init__(self, signals=[], noise_model=None, dtype=np.float32, verbose=False, comps='TQU', singlestream=False):
+        """Initialize a FilterBin Mapmaker for demodulated data 
+        Arguments:
+        * signals: List of Signal-objects representing the models that will be solved
+          jointly for. Typically this would be the sky map and the cut samples. NB!
+          The way the cuts currently work, they *MUST* be the first signal specified.
+          If not, the equation system will be inconsistent and won't converge.
+        * noise_model: A noise model constructor which will be used to initialize the
+          noise model for each observation. Can be overriden in add_obs.
+        * dtype: The data type to use for the time-ordered data. Only tested with float32
+        * verbose: Whether to print progress messages. Not implemented"""
+        if noise_model is None:
+            noise_model = NmatWhite()
+        self.signals      = signals
+        self.dtype        = dtype
+        self.verbose      = verbose
+        self.noise_model  = noise_model
+        self.data         = []
+        self.dof          = MultiZipper()
+        self.ready        = False
+        self.ncomp        = len(comps)
+        self.singlestream = singlestream
+
+    def add_obs(self, id, obs, noise_model=None, deslope=False, split_labels=None):
+        # Prepare our tod
+        ctime  = obs.timestamps
+        srate  = (len(ctime)-1)/(ctime[-1]-ctime[0])
+        if self.singlestream == False:
+            # now we have 3 signals, dsT / demodQ / demodU. We pack them into an array with shape (3,...)
+            tod    = np.array([obs.dsT.astype(self.dtype, copy=False), obs.demodQ.astype(self.dtype, copy=False), obs.demodU.astype(self.dtype, copy=False)])
+            if deslope:
```

Comment:
yes, this is carried over from the first mapmaker written by Sigurd. I think there was no function like this back then. I will remove it from here, but we should also remember to figure out what to do in the ML mapmaker because that also has this same code.